Explorer/Assets/DCL/LOD/Systems/UpdateVisualSceneStateSystem.cs

```diff
@@ -107,20 +107,28 @@ private void IterateOverOneOf<T>(Archetype archetype, ContinuationMethod<T> cont
                     ref PartitionComponent partitionComponent = ref Unsafe.Add(ref partitioncomponentFirstElement, entityIndex);
 
                     if (partitionComponent.IsDirty)
-                    {
                         visualSceneStateResolver.ResolveVisualSceneState(ref visualSceneStateComponent, partitionComponent, sceneDefinitionComponent, lodSettingsAsset, realmData);
 
-                        // we call it directly so we avoid an extra query
-                        if (visualSceneStateComponent.IsDirty)
-                            continuationMethod(entity, ref visualSceneStateComponent, ref sceneDefinitionComponent, ref partitionComponent, ref customComponent);
-                    }
+                    // we call it directly so we avoid an extra query
+                    if (visualSceneStateComponent.IsDirty)
+                        continuationMethod(entity, deltaTime, ref visualSceneStateComponent, ref sceneDefinitionComponent, ref partitionComponent, ref customComponent);
+                    else
+                        visualSceneStateComponent.TimeToChange = 0;
```

Comment:
TO make it clear, what do you think if we reset the time when we mark it as dirty?

Reply:
sounds good :)